src/acp/control.test.ts

```diff
@@ -228,3 +245,745 @@ describe("removeAccessControl", () => {
     expect(getThing(newAccessControlResource, accessControlUrl)).toBeNull();
   });
 });
+
+describe("addPolicyUrl", () => {
+  it("adds the given policy as a regular policy to the given Access Control", () => {
+    const policyUrl = "https://some.pod/policy.ttl#some-policy";
+    const control = createThing();
+
+    const updatedControl = addPolicyUrl(control, policyUrl);
+
+    expect(getIriAll(updatedControl, acp.apply)).toContain(policyUrl);
+  });
+
+  it("does not remove existing policies", () => {
+    const existingPolicyUrl = "https://some.pod/policy.ttl#some-other-policy";
+    const newPolicyUrl = "https://some.pod/policy.ttl#some-policy";
+    const control = setIri(createThing(), acp.apply, existingPolicyUrl);
+
+    const updatedControl = addPolicyUrl(control, newPolicyUrl);
+
+    expect(getIriAll(updatedControl, acp.apply)).toContain(existingPolicyUrl);
+  });
+
+  it("accepts policy URLs as NamedNodes", () => {
+    const policyUrl = "https://some.pod/policy.ttl#some-policy";
+    const control = createThing();
+
+    const updatedControl = addPolicyUrl(
+      control,
+      DataFactory.namedNode(policyUrl)
+    );
+
+    expect(getIriAll(updatedControl, acp.apply)).toContain(policyUrl);
+  });
+
+  it("accepts policy URLs as Things with URLs", () => {
+    const policyUrl = "https://some.pod/policy.ttl#some-policy";
+    const policy = createThing({ url: policyUrl });
+    const control = createThing();
+
+    const updatedControl = addPolicyUrl(control, policy);
+
+    expect(getIriAll(updatedControl, acp.apply)).toContain(policyUrl);
+  });
+});
```

Comment:
There is no tests checking that the input object is left unmodified. I don't think it's a big deal, but this type of test is present in other test suites. Should it be added here ?

This comment applies to other tests in this file.

Reply:
That is cleaner:
- d03f3b5c1e5217a4785b08e8c35b8b05821f26e4
- d5357c62596ad1ca22ebde6f7aa8469a101eb53e